File: app/src/main/java/com/cn29/aac/ui/login/vm/LoginViewModelFactory.java
package com.cn29.aac.ui.login.vm;

import com.cn29.aac.repo.user.AuthRepository;

import javax.inject.Inject;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

/**
 * Created by Charles Ng on 13/10/2017.
 */

public class LoginViewModelFactory extends ViewModelProvider.NewInstanceFactory {

  private AuthRepository authRepository;

  @Inject
  public LoginViewModelFactory(AuthRepository authRepository) {
    this.authRepository = authRepository;
  }


  @Override
  public <T extends ViewModel> T create(Class<T> modelClass) {
    return (T) new LoginViewModel(authRepository);
  }
}

File: app/src/main/java/com/cn29/aac/ui/login/vm/LoginViewModelFactory.kt
package com.cn29.aac.ui.login.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.cn29.aac.repo.user.AuthRepository
import javax.inject.Inject

/**
 * Created by Charles Ng on 13/10/2017.
 */
class LoginViewModelFactory @Inject constructor(private val authRepository: AuthRepository) :
        NewInstanceFactory() {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return LoginViewModel(authRepository) as T
    }

}